### PR TITLE
Adjusting Makefile.one and btfhub.sh with fixes and improvements.

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
bbf7da2 Makefile.one: build dependencies fixes
a8b8507 btfhub.sh: improvement and fixes
bcf796e tracee-make: fix dockerfiles sudoers and dependencies
6ffbad3 Makefile.one: default parallel jobs for libbpf build only

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
@yanivagman this PR contain fixes to Makefile.one logic recently merged. Feel free to merge this, if you agree, or to steal this PR and do any adjustments it might need, as I'll be off for couple of weeks.

**Review by itaysk — COMMENTED** [review]
IMO it's a bit too much that every recipe needs to declare all the commands it will use

**Comment by rafaeldtinoco** [comment]
> IMO it's a bit too much that every recipe needs to declare all the commands it will use

Okay. I can remove checks for basic commands (such as CMD_RM, CMD_MKDIR, CMD_TOUCH, CMD_GREP, CMD_CAT). I'll keep the others, though, as they aren't "installed by default" such as those basic commands.

**Comment by rafaeldtinoco** [comment]
@itaysk I have removed the basic checks now.

**Comment by rafaeldtinoco** [comment]
@itaysk mind giving me a +1 on this one (if you think its good) ? Its blocking PR: #1273 for me (because of rebasing need).

**Comment by itaysk** [comment]
Could someone else please take a look? Don't let me block you

**Comment by rafaeldtinoco** [comment]
@yanivagman could you, please, review this one ? It's a bunch of needed fixes to Makefile.one and I would like to move on with it so I can generate the container images with the new Makefile and, finally, get rid of current building files. 

Thanks.

File: Makefile.one
Comment (yanivagman) [inline Makefile.one:43]: I'm not sure I understand why this is required.
If non-docker make is chosen, does this mean that we create these files on the system of the user?

File: Makefile.one
Comment (yanivagman) [inline Makefile.one:86]: This will break when a newer version of go will be out.
Maybe we should check if version is higher or equal to 1.16 instead?

**Comment by itaysk** [comment]
@grantseltzer @danielpacak PTAL

File: Makefile.one
Comment (rafaeldtinoco) [reply Makefile.one:43]: Those files are created during Makefile execution in current PWD. Without this files, each time the build target is triggered it will be executed. The files mean that the check was done and passed (or else they wouldn't exist). This is a well known workaround for Makefile if you need a logic to be triggered as a check, then you have to bind the logic to the existence of a file that is unmodified between runs. Or else each time you call "make -f Makefile.one tracee-ebpf" the build command will be executed (even if there are no modified files).

File: Makefile.one
Comment (rafaeldtinoco) [reply Makefile.one:86]: I thought about it and decided to bind it to 1.16 or 1.17 because we know, for sure, they are compatible. We don't know if our code is compatible to 1.18 (or 1.19, ...) and the idea was to hard code the supported versions here. With that said, do you still want to make that change ?

Comment (yanivagman) [reply Makefile.one:86]: So you think that go is not backwards compatible? I'm not sure this is true, but I'm also not 100% sure about that

**Review by yanivagman — APPROVED** [review]
LGTM
Please only check about go backwards compatibility

File: Makefile.one
Comment (rafaeldtinoco) [reply Makefile.one:86]: @mtcherni95 any opinion here? Should we do specific versioning tests ? Or assume anything higher than 1.16 would be compatible ?

**Comment by rafaeldtinoco** [comment]
> LGTM Please only check about go backwards compatibility

Thanks. I asked @mtcherni95 opinion before merging this.

File: Makefile.one
Comment (itaysk) [reply Makefile.one:86]: I agree with Yaniv, that we need to check for minimum version (if we use a new language feature not available previously), but not for maximum version (if existing code will stop working), since Go is supposed to be forwards compatible, and in rare cases where it's not it would be a matter of bug fix

**Pull request merged**
